gpkit/constraints/signomial_program.py

```diff
@@ -48,7 +48,7 @@ def __init__(self, cost, constraints, substitutions=None, verbosity=1):
         try:
             _ = self.as_posyslt1()  # should raise an error
             # TODO: is there a faster way to check?
-        except ValueError:
+        except TypeError:
```

Comment:
I think cody has code (calling external solver) that will need to switch from `ValueError` to `TypeError` once this is merged -- correct?